app/assets/src/components/views/amr_heatmap/AMRHeatmapView.jsx

```diff
@@ -50,28 +55,39 @@ export default class AMRHeatmapView extends React.Component {
 
   componentDidMount() {
     const { sampleIds } = this.props;
-    this.requestAMRCountsData(sampleIds);
+    this.requestSampleData(sampleIds);
   }
 
-  async requestAMRCountsData(sampleIds) {
+  async requestSampleData(sampleIds) {
     const rawSampleData = await getAMRCounts(sampleIds);
     const filteredSamples = rawSampleData.filter(
       sampleData => sampleData.error === ""
     );
-    const samplesWithAMRCounts = this.processAMRCounts(filteredSamples);
+    const samplesWithKeyedMetadata = filteredSamples.map(sample =>
+      Object.assign({}, sample, {
+        metadata: processMetadata(sample.metadata, true),
+      })
+    );
+    const samplesWithAMRCounts = this.processSampleAMRCounts(
+      samplesWithKeyedMetadata
+    );
     const maxValues = this.findMaxValues(samplesWithAMRCounts);
+    const rawSamplesMetadataTypes = await getSampleMetadataFields(
```

Comment:
I would recommend using `Promise.all` to do the two HTTP requests in parallel. You can see examples in other files. This means less time waiting on the network.